src/dune_rules/mdx.ml

```diff
@@ -247,7 +256,8 @@ let files_to_mdx t ~sctx ~dir =
   in
   let must_mdx src_path =
     let file = Path.Source.basename src_path in
-    Predicate_lang.Glob.exec t.files ~standard:default_files file
+    let standard = default_files_of_version t.loc t.version in
+    Predicate_lang.Glob.exec t.files ~standard file
```

Comment:
It's weird that we're doing this in 2 places. The way I understand it:
- at parsing time, we default to `default_files`
- here, we're saying that `:standard` evaluates to `default_files`

I think that we can just do that here, and at parse time default to `Standard` in the files field. (this can be done in a separate PR, or here)

Reply:
as a bonus this should make the parser applicative again (no `let*`)

Reply:
Was already wondering why this happened in two different places, but `:standard` explains it - indeed, it is nicer to just fill it with `Predicate_lang.Standard` then parsing and resolve it to the globs in the evaluation. Added it in this PR and also extended the test to show that passing `(files :standard)` processes `*.mld` files as well.